src/js/jsx/shared/Droppable.jsx

```diff
@@ -26,128 +26,142 @@ define(function (require, exports, module) {
+        
+    var droppableCounter = 0;
 
-    /**
-     * Create a composed Droppoable component
-     * 
-     * The getProps function should return an object of the form:
-     *  {
-     *     key: (Unique key for this droppable),
-     *     keyObject: (Object that is being dropped on),
-     *     validate: (Function that accepts an argument of 
-     *         object being dropped on this keyObject, returns a boolean),
-     *     handleDrop: (Function to handle dropping of object on this droppable)
-     *    }
-     *
-     * @param {ReactComponent} Component Component to wrap
-     * @param {function} getProps function to return an object with the props required by Droppable
-     * @param {function?} isEqual
-     * @param {function?} shouldUpdate
-     * @return {ReactComponent}
-     */
-    var createWithComponent = function (Component, getProps, isEqual, shouldUpdate) {
-        var Droppable = React.createClass({
-            mixins: [FluxMixin],
+    var Droppable = React.createClass({
+        mixins: [FluxMixin],
+        
+        propTypes: {
+            // Type of Draggable that is accepted
+            accept: React.PropTypes.string.isRequired,
 
             /**
-             * Register the droppable component with the draganddrop store.
+             * @callback Draggable~handleDragTargetEnter
+             * @param {Immutable.List.<object>} draggedTargets
+             * @param {{x: number, y:number}} position
              */
-            _register: function () {
-                var flux = this.getFlux(),
-                    options = getProps(this.props),
-                    zone = options.zone,
-                    droppable = {
-                        node: React.findDOMNode(this),
-                        key: options.key,
-                        keyObject: options.keyObject,
-                        isValid: options.isValid,
-                        onDrop: options.handleDrop,
-                        droppable: this
-                    };
-
-                flux.store("draganddrop").registerDroppable(zone, droppable);
-            },
-
-            shouldComponentUpdate: shouldUpdate,
-
+            handleDragTargetEnter: React.PropTypes.func,
+            
             /**
-             * Get the droppable's registration information.
-             *
-             * Returned object has properties:
-             *     key: (Unique key for this droppable),
-             *     keyObject: (Object that is being dropped on),
-             *     node: (DOM element for this droppable),
-             *     isValid: (Function that accepts an argument of 
-             *         object being dropped on this keyObject, returns a boolean),
-             *     handleDrop: (Function to handle dropping of object on this droppable)
-             *
-             * @return {Array.<object>} with information about registration for easy ingestion into OrderedMap
+             * @callback Draggable~handleDragTargetMove
```

Comment:
This is a way of documenting callbacks, where X is the package name, Y is the callback name, and together with ~ is the name to refer to this callback in `@param` and `@type`. (e.g.  @param {Draggable~handleDragTargetMove}) http://usejsdoc.org/tags-callback.html  I found this is the best way (by far) to document a callback, as it:

* help avoid spreading duplicate jsdoc across the same type of callbacks. Before we have to write the same @param list repeatedly for each callback; now it can be replaced by @type {X~Y})
* provides a central location for the update-to-date jsdoc of the callback.
* easier to locate callbacks with the same type (which means easier/safer for refactoring/bug fixing).